server.py

```diff
@@ -0,0 +1,218 @@
+    def set_security(self, security):
+        """Attribute Setter for whether data is encrypted"""
+        if security in SecurityLevel:
+            self._security = security
+        else:
+            raise Exception("Invalid security level")
+
+    @staticmethod
+    def get_default_dir():
+        """Returns default directory for saving textfiles"""
+        current = os.path.dirname(os.path.realpath(__file__))
+        output = os.path.join(current, "Output")
+        return output
+
+    def get_output_directory(self):
+        """Attribute Getter for Output Directory"""
+        return self._output_directory
+
+    def set_output_directory(self, directory):
+        """Attribute Setter for the directory where textfiles will be written"""
+        # Ensure Directory exists
+        if not os.path.isdir(directory):
+            os.mkdir(directory)
+        # Set Attribute
+        self._output_directory = directory
+
+    def set_destination(self, destination):
+        """Attribute Setter for destination"""
+        if destination in ServerDestination:
+            self._destination = destination
+        else:
+            raise Exception("Invalid Server Destination")
+
+    def set_source(self, source):
+        """Attribute Setter for source"""
+        self.pickler.set_source_type(source)
+
+    def set_pickling_format(self, pickling_format):
+        """Attribute Setter for the Pickling Format"""
+        self.pickler.set_pickling_format(pickling_format)
+
+    @staticmethod
+    def get_server_host():
+        """Returns the hostname of the server machine for passing to client"""
+        return socket.gethostname()  # Local Machine Name
+
+    def _get_socket(self):
+        """Returns a socket"""
+        host = socket.gethostname()  # Local Machine Name
+        my_sock = socket.socket()  # Create a socket object
+        port = self.PORT_NUMBER  # reserve a port
+        my_sock.bind((host, port))  # bind the socket to the port of the machine
+        return my_sock
+
+    def listen(self):
+        """Sets the socket listening for clients"""
+        sock = self._get_socket()
+        sock.listen(self.BACKLOG)  # wait for client connection
+
+        while True:
+            # Establish connection with client
+            conn, addr = sock.accept()
+            print("Got connection from", addr)
+
+            # Receive data from client
+            data = conn.recv(self.BUFFER_SIZE)
+
+            # Filter data for message or content
+            data = self.filter_data(data, conn)
+
+            if data is not None:
+                print("Server received", repr(data))
+                self._received_data = data
+                self.decision_tree()
+                conn.send(Conversation.DataReceived)
+
+            conn.close()
+
+    def filter_data(self, message, connection):
+        """Is the incoming data a message or data?"""
+
+        # Is this a message? If so perform action
+        if message in Conversation:

```

Comment:
When we move to a multi-threaded version, much of the server class code will move to a connection class.
However, I'm assuming that anything might arrive at the server, at any time and in any order. The logic here is to tell whether the incoming data is a message (and if so what it is) or if it is a dictionary or text file, and respond appropriately, rather than relying on communications happening in the choreographed order.